techbuzz/src/main/resources/templates/fragments/post.html

```diff
@@ -4,59 +4,75 @@
+                Posted By: <a href="#" class="text-dark"  th:href="@{'/userSpecific/profile/'+ ${post.createdBy.id}}" th:text="${post.createdBy.name}">Name</a>
+                Date: <span class="text-dark" th:text="${#temporals.format(post.createdAt, 'dd-MMM-yyyy')}">Date</span>
+                <span th:if ="${type=='userSpecificPost'}" >Category: <span class="text-dark" th:text="${post.category.name}">Category</span></span>
+            </div>
+            <div th:text="${post.content}">
+                Post content
+            </div>
+            <p th:if ="${type=='normalPost'}" sec:authorize="isAuthenticated()" class="mt-3 mb-0">
 
-                    <a th:if="${post.editable}" class="btn btn-primary" href="#" th:href="${'/posts/'+post.id+'/edit'}">
-                        <i class="fas fa-edit"></i> Edit
-                    </a>
+                <a th:if="${post.editable} " class="btn btn-primary" href="#" th:href="${'/posts/'+post.id+'/edit'}">
+                    <i class="fas fa-edit"></i> Edit
+                </a>
 
-                    <button th:if="${post.editable}" class="btn btn-danger"
+                <button th:if="${post.editable} " class="btn btn-danger"
                         th:onclick="'javascript:deletePost( \''+${post.id}+'\');'">
-                        <i class="fas fa-trash-alt"></i> Delete
-                    </button>
+                    <i class="fas fa-trash-alt"></i> Delete
+                </button>
 
-                    <button class="btn btn-success" th:classappend="${post.upVoted()} ? disabled : ''"
+                <button class="btn btn-success" th:classappend="${post.upVoted()} ? disabled : ''"
                         th:onclick="'javascript:addVote( \''+${post.id}+'\', 1);'">
-                        <span th:text="${post.getUpVoteCount()}" class="p-2">1</span>
-                        <i class="fa-solid fa-thumbs-up"></i>
-                    </button>
+                    <span th:text="${post.getUpVoteCount()}" class="p-2">1</span>
+                    <i class="fa-solid fa-thumbs-up"></i>
+                </button>
 
-                    <button class="btn btn-danger" th:classappend="${post.downVoted()} ? disabled : ''"
+                <button class="btn btn-danger" th:classappend="${post.downVoted()}  ? disabled : ''"
                         th:onclick="'javascript:addVote( \''+${post.id}+'\', -1);'">
-                        <span th:text="${post.getDownVoteCount()}" class="p-2">1</span>
-                        <i class="fa-solid fa-thumbs-down"></i>
-                    </button>
-                </p>
-                <p sec:authorize="!isAuthenticated()" class="mt-3 mb-0">
+                    <span th:text="${post.getDownVoteCount()}" class="p-2">1</span>
+                    <i class="fa-solid fa-thumbs-down"></i>
+                </button>
+            </p>
+            <p th:if ="${type=='normalPost'}" sec:authorize="!isAuthenticated()" class="mt-3 mb-0">
```

Comment:
I am wondering is it better to use same template (post.html) and add conditional rendering `th:if ="${type=='normalPost'}"` or created a separate template. Personally, I prefer to create a separate template because a little bit of code duplication is better than a single but complicated template.

Reply:
Created a separate template